pandas/core/arrays/integer.py

```diff
@@ -293,6 +295,43 @@ def __array__(self, dtype=None):
         """
         return self._coerce_to_ndarray()
 
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        out = kwargs.get('out', ())
+
+        for x in inputs + out:
+            if not isinstance(x, self._HANDLED_TYPES + (IntegerArray,)):
+                return NotImplemented
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not NotImplemented:
+            return result
+
+        if (method == '__call__'
+                and ufunc.signature is None
+                and ufunc.nout == 1):
+            # only supports IntegerArray for now
+            args = [a._data for a in inputs]
+            masks = [a._mask for a in inputs]
+            result = ufunc(*args, **kwargs)
+            mask = np.logical_or.reduce(masks)
+            if result.dtype.kind in ('i', 'u'):
+                return IntegerArray(result, mask)
+            else:
+                result[mask] = np.nan
+                return result
+
+        # fall back to array for other ufuncs
```

Comment:
should this be a generic function that one can just call  (e.g. put it in ops.py and pass in self, ufuc, method, inputs, kwargs)? IOW is this called anywhere else